Bug: Import Failure on Python 3.12

### Describe the bug

I was testing Geomstats on python 3.12 and got the following error. Not sure if full support has been added for python 3.12 yet but see that no bug issue has been raised for this. 

### Steps/Code to Reproduce

```
from geomstats.geometry.euclidean import Euclidean
```

### Expected Behaviour

Example: No error is thrown

### Actual Behaviour

```
from geomstats.geometry.euclidean import Euclidean
/opt/hostedtoolcache/Python/3.12.1/x64/lib/python3.12/site-packages/geomstats/geometry/euclidean.py:3: in <module>
    import geomstats.backend as gs
/opt/hostedtoolcache/Python/3.12.1/x64/lib/python3.12/site-packages/geomstats/__init__.py:7: in <module>
    import geomstats.geometry.discrete_curves
/opt/hostedtoolcache/Python/3.12.1/x64/lib/python3.12/site-packages/geomstats/geometry/discrete_curves.py:11: in <module>
    import geomstats.backend as gs
E   ModuleNotFoundError: No module named 'geomstats.backend'
```

### Your environment

```shell
Geomstats==2.7.0
Python 3.12
OS: Windows, Ubuntu, & MacOS
```

--- a/geomstats/_backend/__init__.py
+++ b/geomstats/_backend/__init__.py
@@ -215,7 +215,8 @@ class BackendImporter:
     """Importer class to create the backend module."""
 
     def __init__(self, path):
-        self._path = path
+        self._path = self.name = path
+        self.loader = self
 
     @staticmethod
     def _import_backend(backend_name):
@@ -289,6 +290,10 @@ class BackendImporter:
 
         logging.debug(f"geomstats is using {BACKEND_NAME} backend")
         return module
+
+    def find_spec(self, fullname, path=None, target=None):
+        """Find module."""
+        return self.find_module(fullname, path=path)
 
 
 sys.meta_path.append(BackendImporter("geomstats.backend"))

